src/Uno.Toolkit.UI/Extensions/ResourceDictionaryExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+#else
+using Windows.UI.Xaml;
+#endif
+
+namespace Uno.Toolkit.UI;
+
+internal static class ResourceDictionaryExtensions
+{
+	public static ResourceDictionary DeepClone(this ResourceDictionary rd)
+	{
+		if (rd.Source is not null) return new ResourceDictionary() { Source = rd.Source };
+
+		var result = new ResourceDictionary();
+
+		if (rd.ThemeDictionaries is { })
+		{
+			foreach (var (key, value) in rd.ThemeDictionaries)
+			{
+				result.ThemeDictionaries[key] = (value as ResourceDictionary)?.DeepClone() ?? value;
+			}
+		}
+		if (rd.MergedDictionaries is { })
+		{
+			foreach (var md in rd.MergedDictionaries)
+			{
+				result.MergedDictionaries.Add(md.DeepClone());
+			}
+		}
+		foreach (var (key, value) in rd)
+		{
+			result[key] = value;
+		}
+
+		return result;
+	}
+}
```

Comment:
```suggestion
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
#if IS_WINUI
using Microsoft.UI.Xaml;
#else
using Windows.UI.Xaml;
#endif

namespace Uno.Toolkit.UI
{
	internal static class ResourceDictionaryExtensions
	{
		/// <summary>
		/// Creates a deep clone of the given <see cref="ResourceDictionary"/>.
		/// </summary>
		/// <param name="rd">The resource dictionary to clone.</param>
		/// <returns>A deep cloned copy of the resource dictionary.</returns>
		public static ResourceDictionary DeepClone(this ResourceDictionary rd)
		{
			try
			{
				// If the source URI is not null, create a new ResourceDictionary with the same source URI.
				if (rd.Source is not null)
				{
					return new ResourceDictionary() { Source = rd.Source };
				}

				// Initialize a new ResourceDictionary to hold the cloned values.
				var result = new ResourceDictionary();

				// Clone theme dictionaries, if any.
				CloneThemeDictionaries(rd, result);
				
				// Clone merged dictionaries, if any.
				CloneMergedDictionaries(rd, result);
				
				// Clone all key-value pairs from the original ResourceDictionary.
				CloneResources(rd, result);

				return result;
			}
			catch (Exception ex)
			{
				// Log the error appropriately.
				System.Diagnostics.Debug.WriteLine($"Error during ResourceDictionary deep cloning: {ex.Message}");
				throw;
			}
		}

		/// <summary>
		/// Clones the theme dictionaries from the source to the target ResourceDictionary.
		/// </summary>
		/// <param name="source">The source ResourceDictionary.</param>
		/// <param name="target">The target ResourceDictionary.</param>
		private static void CloneThemeDictionaries(ResourceDictionary source, ResourceDictionary target)
		{
			if (source.ThemeDictionaries is {})
			{
				foreach (var (key, value) in source.ThemeDictionaries)
				{
					// Recursively deep clone theme dictionaries, if they exist.
					target.ThemeDictionaries[key] = (value as ResourceDictionary)?.DeepClone() ?? value;
				}
			}
		}

		/// <summary>
		/// Clones the merged dictionaries from the source to the target ResourceDictionary.
		/// </summary>
		/// <param name="source">The source ResourceDictionary.</param>
		/// <param name="target">The target ResourceDictionary.</param>
		private static void CloneMergedDictionaries(ResourceDictionary source, ResourceDictionary target)
		{
			if (source.MergedDictionaries is {})
			{
				foreach (var md in source.MergedDictionaries)
				{
					// Recursively deep clone merged dictionaries.
					target.MergedDictionaries.Add(md.DeepClone());
				}
			}
		}

		/// <summary>
		/// Clones the resources (key-value pairs) from the source to the target ResourceDictionary.
		/// </summary>
		/// <param name="source">The source ResourceDictionary.</param>
		/// <param name="target">The target ResourceDictionary.</param>
		private static void CloneResources(ResourceDictionary source, ResourceDictionary target)
		{
			foreach (var (key, value) in source)
			{
				// Copy each key-value pair from the source to the target dictionary.
				target[key] = value;
			}
		}
	}
}
```

Reply:
Suggestion to better understand the logic in my opinion

Reply:
i dont think there is much values in extracting those 2/3lines blocks
but i will keep the xml comment, and try-catch